static-instrumenter/agent-instrumenter/src/test/java/io/opentelemetry/contrib/staticinstrumenter/ClassArchiveTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.staticinstrumenter;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.jar.JarFile;
+import java.util.zip.ZipOutputStream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class ClassArchiveTest {
+
+  @TempDir private File tempDir;
+
+  @Test
+  void shouldCopyAllClasses(@TempDir File destination) throws IOException {
+
+    // given
+    byte[] newTransformed = new byte[] {(byte) 0xCA, (byte) 0xFE, (byte) 0xBA, (byte) 0xBE};
+    Map<String, byte[]> transformed = Collections.singletonMap("test/TestClass", newTransformed);
+    String jar = JarTestUtil.createJar(tempDir, "one.jar", "test/TestClass.class", "second.class");
```

Comment:
It probably makes sense to commit JAR files into `resources` for these tests instead of slowing down the tests by creating the same JAR every time

Reply:
Reminder of this comment

Reply:
Yep, another point where I would prefer to leave things as they're now. My reasons:
1. Execution time for tests in contrib is not that important (especially diff made by creating several files on disk)
2. If the JARs are committed, list of classes (that needs to be verified after copy) will need to be hardcoded - tests will be harder to follow (understand)
3. I'm used to creating temp files in tests as a good testing practice

Reply:
FWIW, I found the current pattern to be hard to understand. Since the code's main job is to create a JAR, it was surprising to see the test also creating a jar. After a while I realized that the created JARs are input for processing - it was not obvious these are test "input constants". I think checked in JARs will make this more obvious and make the tests much easier to follow. A rule of thumb is the less preparation logic in a test, the simpler and easier to read the test is.

Reply:
As U wish then ;)